flagsmith-core.ts

```diff
@@ -173,22 +177,38 @@ const Flagsmith = class {
         };
 
         if (identity) {
+            if (isIdentityConfig(identity)) {
+                var {identifier, transient} = identity;
```

Comment:
const or let, never var

Reply:
Okay, but how do I access these from inside the promise later?

(Sorry for being a JS noob)

Reply:
No worries! var hoists, so we should hoist with let

```
let identifier: string; // Adjust the type according to your actual type
let transient: boolean | undefined = false;

if (isIdentityConfig(identity)) {
    ({identifier, transient} = identity);
} else {
    identifier = identity;
}
```

Reply:
Nice, this is done.